src/main/java/edu/oregonstate/cs361/battleship/Main.java

```diff
@@ -182,137 +143,114 @@ public static String placeShipUpdated(Request req) {
-                    if (id.toLowerCase().contains("aircraftcarrier")) {
-                        CAircraftCarrier.setStart(computerStart.getAcross(), computerStart.getDown());
-                        CAircraftCarrier.setEnd(computerEnd.getAcross(), computerEnd.getDown());
-
-                    } else if (id.toLowerCase().contains("battleship")) {
-                        CBattleship.setStart(computerStart.getAcross(), computerStart.getDown());
-                        CBattleship.setEnd(computerEnd.getAcross(), computerEnd.getDown());
 
-                    } else if (id.toLowerCase().contains("submarine")) {
-                        CSubmarine.setStart(computerStart.getAcross(), computerStart.getDown());
-                        CSubmarine.setEnd(computerEnd.getAcross(), computerEnd.getDown());
-
-                    } else if (id.toLowerCase().contains("dinghy")) {
-                        CDinghy.setStart(computerStart.getAcross(), computerStart.getDown());
-                        CDinghy.setEnd(computerEnd.getAcross(), computerEnd.getDown());
+                if (stop == 1) {
+                    for (movingPoint = movingPoint - 1; movingPoint < (computer_length_increase + 2); movingPoint++) {
+                        if (horizontal == 1) {
+                            cord.setAcross(movingPoint);
+                            cord.setDown(stoppedPoint);
+
+                        } else {
+                            cord.setAcross(stoppedPoint);
+                            cord.setDown(movingPoint);
+                        }
+
+                        //if ship lands on another ship then
+                        for (int i = 0; i < 5; i++) {
+                            if (Hit(CArray[i].getStart(), CArray[i].getEnd(), cord)) {
+                                stop = 0;
+                            }
+                        }
+                    }
 
-                    } else if (id.toLowerCase().contains("clipper")) {
-                        CClipper.setStart(computerStart.getAcross(), computerStart.getDown());
-                        CClipper.setEnd(computerEnd.getAcross(), computerEnd.getDown());
+                    if (stop == 1) {
+                        model.getShipByID(CArray[id].getName()).setStart(computerStart.getAcross(), computerStart.getDown());
+                        model.getShipByID(CArray[id].getName()).setEnd(computerEnd.getAcross(), computerEnd.getDown());
+                        CArray = model.resetArrayUpdated(model, false);
                     }
                 }
             }
         }
 
-        Gson gson = new Gson();
-        return gson.toJson(model);
+        return model;
 
     }
 
+    public static BattleshipModelUpdated placeShipUpdatedEasy(BattleshipModelUpdated model) {
 
+        for(int id = 0; id < 5; id++) {
+            Ship[] CArray = model.resetArrayUpdated(model, false);
+            int computerLength = model.getShipByID(CArray[id].getName()).getLength() - 1;
```

Comment:
Take Aircraftcarrier for instance if it starts at (5,5) and goes horizontally then it would be placed 5 + (length of ship) 5 = 10 so that means (5,5) (6,5) (7,5) (8,5) (9,5) (10,5) which is length of 6 and so if you do do -1 then the ship will be placed on (5,5) (6,5) (7,5) (8,5) (9,5) which is length on 5.